src/openms/include/OpenMS/ANALYSIS/TOPDOWN/DeconvolutedSpectrum.h

```diff
@@ -0,0 +1,144 @@
+
+    /// move constructor
+    DeconvolutedSpectrum(DeconvolutedSpectrum &&other) = default;
+
+    /// assignment operator
+    DeconvolutedSpectrum &operator=(const DeconvolutedSpectrum &fd) = default;
+
+    /**
+        @brief write the header in the output file (spectrum level)
+        @param fs file stream to the output file
+        @param n the index to the mass
+        @param detail if set true, detailed information of the mass (e.g., peak list for the mass) is written
+   */
+    static void writeDeconvolutedMassesHeader(std::fstream &fs, int &n, bool detail);
+
+    /**
+      @brief write the deconvoluted masses in the output file (spectrum level)
+      @param fs file stream to the output file
+      @param param FLASHDeconv paramter
+    */
+    void writeDeconvolutedMasses(std::fstream &fs,
+                                 const String &fileName,
+                                 const FLASHDeconvHelperStructs::PrecalculatedAveragine &avg,
+                                 bool writeDetail);
+
+    /**
+      @brief write the deconvoluted masses TopFD format
+      @param fs file stream to the output file
+      @param id the index to the spectrum. updated outside.
+ */
+    void writeTopFD(std::fstream &fs, int id, const FLASHDeconvHelperStructs::PrecalculatedAveragine &avg);
+
+    /// cast DeconvolutedSpectrum into MSSpectrum object to write mzml format
+    MSSpectrum toSpectrum(int charge);
+
+    /// write the header for Thermo Inclusion List header format
+    static void writeThermoInclusionHeader(std::fstream &fs);
+
+    /// for memory save... clear unnecessary information in mass tracing
+    void clearPeakGroupsChargeInfo();
+
+    /**
+     @brief register the precusor info in this from the precursor DeconvolutedSpectrum
+     @param precursorSpectrum the precursor DeconvolutedSpectrum
+     */
+    bool registerPrecursor(DeconvolutedSpectrum &precursorSpectrum);
+
+    /// original spectrum setter
+    MSSpectrum &getOriginalSpectrum();
+
+    /// peakGroup getter
+    PeakGroup getPrecursorPeakGroup();
+
+    /// precursor charge getter : set in registerPrecursor
+    int getPrecursorCharge();
+
+    /// get max mass - which is min mass between max mass specified by users or mass determined by precursor mass for MSn
+    double getCurrentMaxMass(double maxMass);
+
+    /// get max charge - which is min charge between max charge specified by users or charge determined by precursor charge for MSn
+    int getCurrentMaxCharge(int maxCharge);
+
+  private:
+    /// the original spectrum from which this is generated
+    MSSpectrum spec;
+    /// precursor peakGroup (or mass)
+    PeakGroup *precursorPeakGroup = nullptr;
+    /// precursor peak (not deconvoluted one)
+    Precursor precursorPeak;
+    /// activation method for file output
+    std::string activationMethod;
+    /// scan number and precursor scan number
+    int scanNumber, precursorScanNumber;
```

Comment:
- rename to scan_number_ and precursor_scan_number_
- is precursor_scan_number_ redundant?